### ISSUE TITLE
Loading a .ply with sh=2 seems to only render with sh=0

### ISSUE THREAD

**Issue by vincentwoo** [playcanvas/supersplat#359]
I trained a splat with SH bands up to f_rest_23 in the .ply, but in supersplat I'm not seeing any spherical harmonics. The slider still goes from 0-3, but there is no change between them. I can provide an example ply if that helps but I imagine this is an easy fix somewhere, just logging it here for now.

**Comment by slimbuck** [comment]
Hi @vincentwoo , yes uncompressed PLY assumes either 0 or 3 bands of SH. Thanks for logging this issue, we'll address it soon as we can.

**Comment by arcman7** [comment]
1. Where does the code decide how many sh bands to use?

2. What is the right way to express that I want to render a given .ply model with 3 sh bands?

**Comment by slimbuck** [comment]
If you're generating the data yourself, then you could just add 0's for band 3 for now?

**Comment by vincentwoo** [comment]
Oh, yeah, I can work around it, just thought it might be nice to support. I think a lot of scenes actually train totally fine with lower spherical harmonics (i.e. do not require high frequency view-dependent effects) and it would be nice if it worked out of the box.

**Comment by slimbuck** [comment]
Sorry I missed your comment @arcman7:

> Where does the code decide how many sh bands to use?

The underlying playcanvas engine does a check on number of spherical harmonics and either renders them all if 45 coefficients are found or it renders without spherical harmonics. This has been fine up until now, but needs to be fixed.

[This](https://github.com/playcanvas/engine/blob/main/src/scene/gsplat/gsplat-data.js#L325) is the test for presence of sh data, which is used to pack the data [here](https://github.com/playcanvas/engine/blob/main/src/scene/gsplat/gsplat.js#L93).

> What is the right way to express that I want to render a given .ply model with 3 sh bands?

Just make sure there are 45 coefficients of spherical harmonics. 

I will update the engine sooner rather than later so we correctly support 0, 1, 2 and 3 bands.

NOTE that compressed.ply format does support 0, 1, 2, 3 bands already.

**Comment by slimbuck** [comment]
Do you have any 1 or 2 band scenes I could use for testing?

**Comment by slimbuck** [comment]
Actually I just changed our loader to drop extra bands for testing. Submitted a PR to the engine to support this https://github.com/playcanvas/engine/pull/7248

**Comment by arcman7** [comment]
> Do you have any 1 or 2 band scenes I could use for testing?

I'd be interested in testing this as well. The few attempts myself and @jihan1218 made to train a scene on SH deg anything other than 3 didn't work out well, I've reached out to see if anyone has viable test scenes in SH 1 & 2.

**Comment by vincentwoo** [comment]
sure, here's a SH=2 scene I trained a while back https://www.dropbox.com/scl/fi/5hmhlnonj3zmlpfvkchdg/sutro_2m_splats_SH2.ply?rlkey=wfm9xnjude93ey6ks2pdh72ny&dl=0

**Comment by arcman7** [comment]
@vincentwoo thank you for providing that file to test with! Just got back from CES, so I'll have some time this week to tinker with it.

**Comment by arcman7** [comment]
@slimbuck do each of your merged PRs into the main branch in turn trigger a new release with the playcanvas npm package? How should I be pulling in your latest changes into my own projects?

**Comment by arcman7** [comment]
From what I can tell with that particular file, the rendering seems to work as expected, but I cannot clearly determine if the view dependent coloring is working correctly or not. A smaller scene would help, I'll try to do some more testing soon this weekend.

**Comment by vincentwoo** [comment]
I believe this works in supersplat now!

**Comment by arcman7** [comment]
Yes - we just started generating SH=2 splat scenes and can confirm on our end this works as well.
